kde/qtcurve/Portfile

```diff
@@ -24,63 +29,224 @@ maintainers         gmail.com:rjvbertin
                     http://craigd.wikispaces.com/file/view/
-#distname            ${name}-KDE4-${version}
 
 livecheck.type      regex
 livecheck.url       http://quickgit.kde.org/?p=qtcurve.git
 livecheck.regex     (\\d+(\\.\\d+)+)
 
-depends_lib-append  port:gtk2 \
-                    port:kdelibs4 \
-                    port:qt4-mac
+worksrcdir          ${name}-${qtc_version}
 
-configure.args-append   -DQTC_QT4_STYLE_SUPPORT:BOOL=OFF -DENABLE_QT5:BOOL=OFF \
-                        -DQTC_QT4_ENABLE_KWIN:BOOL=OFF
+PortGroup           muniversal 1.0
 
-variant x11 description {build x11 support} {}
-if { [variant_isset x11] } {
-	depends_lib-append port:xorg-libX11
-	configure.args-append -DQTC_ENABLE_X11:BOOL=ON
-} else {
-	configure.args-append -DQTC_ENABLE_X11:BOOL=OFF
-}
+test.run            yes
 
-variant gtk2 description {build the gtk2 plugin} {}
-if { [variant_isset gtk2] } {
-	depends_lib-append port:gtk2
-	configure.args-append -DENABLE_GTK2:BOOL=ON
-} else {
-	configure.args-append -DENABLE_GTK2:BOOL=OFF
+# This is a guard against installing a port variant for the still uncommitted KF5
+if {${subport} eq "${name}-qt5"} {
+    if {![variant_isset qtonly]} {
+        default_variants +qtonly
+    }
 }
 
-variant qt5 description {build for qt5 too} {}
-if { [variant_isset qt5] } {
-	depends_lib-append port:qt5-mac
-	# default for ENABLE_QT5=on:
-	configure.args-delete -DENABLE_QT5:BOOL=OFF
+subport ${name}-qt5 {
+    if {![variant_isset qtonly]} {
+        long_description        A highly configurable widget style for Qt5/KF5
+    } else {
+        long_description        A highly configurable widget style for Qt5
+    }
 }
 
-variant devel description {Prevent stripping of binaries} {}
-if { [variant_isset devel] } {
-	configure.args-delete -DCMAKE_BUILD_TYPE=Release
-	configure.args-append -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo -DCMAKE_STRIP:FILEPATH=/bin/echo
+if {${os.platform} eq "darwin" && ${os.major} <= 10} {
+    variant libcxx description {build against libc++ (untested)} {
+        PortGroup           cxx11 1.0
+    }
+    if {![variant_isset libcxx]} {
+        depends_build-delete    port:clang-3.4 port:clang-3.5
+        ui_msg "This port only builds with configure.compiler=macports-gcc-4.7 (from port:gcc47) or newer on OS X 10.6"
+    }
 }
 
-variant qtonly description {Build only Qt style support, for installations without KDE} {}
-if { [variant_isset qtonly] } {
-	depends_lib-delete    port:kdelibs4
-#	configure.args-append -DQTC_QT_ONLY:BOOL=ON
-	configure.args-append -DQTC_QT4_ENABLE_KDE:BOOL=OFF
-} else {
+subport ${name}-gtk2 {
+    PortGroup               cmake 1.1
+    cmake.out_of_source     yes
+    long_description        The GTk2 version of the highly configurable QtCurve widget style for Qt
+
+    post-extract            { file mkdir ${workpath}/build }
+    post-patch {
+        // QtCurve-gtk2 has pure and optional runtime dependencies on KDE4 (and KF5)
```

Comment:
@RJVB: I wasn't aware of that TCL also knows the double-slash as a comment. Interesting. :)